intezer_sdk/operation.py

```diff
@@ -41,10 +36,58 @@ def wait_for_completion(self, interval: int = None, sleep_before_first_check=Fal
             time.sleep(interval)
             operation_result = self._api.get_url_result(self.url)
 
-        self.status = AnalysisStatusCode.FINISH
+        self.status = OperationStatusCode.FINISH
         self.result = operation_result.json()['result']
 
 
+class Operation(BaseOperation):
+    def get_result(self):
+        if self.status != OperationStatusCode.FINISH:
+
+            operation_result = self._api.get_url_result(self.url)
+
+            if handle_response_status(operation_result.status_code):
+                self.result = operation_result.json()['result']
+                self.status = OperationStatusCode.FINISH
+            else:
+                raise errors.OperationStillRunning('operation')
+        return self.result
+
+
+class PaginatedOperation(BaseOperation):
```

Comment:
add `__iter__` implementation for simplicity so the user can just iterate it over a for loop